main.go

```diff
@@ -122,9 +123,15 @@ func (s *Server) ServeHTTP(res http.ResponseWriter, req *http.Request) {
 
 			res.Header().Set("Content-Type", "text/html")
 			res.WriteHeader(http.StatusInternalServerError)
-			_, _ = res.Write([]byte(fmt.Sprintf(`
-	<p>Error building URL with %s: %#v</p>
-			`, hostname, err)))
+			data := struct {
+				Hostname string
+				Error    error
+			}{
+				Hostname: hostname,
+				Error:    err,
+			}
+			t, _ := template.New("url building error").Parse("<p>Error building URL with {{.Hostname}}: {{.Error}}</p>")
```

Comment:
Could these templates be declared with `template.Must()` instead to declare these as global variables? It'd be good to catch unrecoverable parsing failures at start-up time, rather than suppressing any returned `error`s.

Relatedly, we should catch `t.Execute()` non-nil errors and respond using something like `http.Error(res, err.Error(), http.StatusInternalServerError)` in those cases